pyiceberg/table/metadata.py

```diff
@@ -411,6 +414,21 @@ def new_table_metadata(
     if table_uuid is None:
         table_uuid = uuid.uuid4()
 
+    format_version = properties.get("format-version", DEFAULT_FORMAT_VERSION)
+    if format_version == "1":
```

Comment:
It looks like were mixing types here.
```suggestion
    format_version = int(properties.get("format-version", DEFAULT_FORMAT_VERSION))
    if format_version == 1:
```